AccessHandle.md

```diff
@@ -0,0 +1,206 @@
+creation of access handles, but not of other writable streams. This prevents
+the file from being modified from multiple contexts, while still being
+backwards compatible with the current OPFS spec and supporting multiple
+*WritableFileStreams* at once.
+
+Creating a [File](https://www.w3.org/TR/FileAPI/#dfn-file) through *getFile()*
+would be possible when a lock is in place. The returned File behaves as it
+currently does in OPFS i.e., it is invalidated if file contents are changed
+after it was created. It is worth noting that these Files could be used to
+observe changes done through the new API, even if a lock is still being held.
+
+## Open Questions
+
+### Naming
+
+The exact name of the new methods hasn’t been defined. The current placeholder
+for data access is *createAccessHandle()* and *createSyncAccessHandle()*.
+*createUnflushedStreams()* and *createDuplexStream()* have been suggested.
+
+### Assurances on non-awaited consistency
+
+It would be possible to clearly specify the behavior of an immediate async read
+operation after a non-awaited write operation, by serializing file operations
+(as is currently done in Storage Foundation API). We should decide if this is
+convenient, both from a specification and performance point of view.
+
+## Appendix
+
+### AccessHandle IDL
+
+```webidl
+interface FileSystemFileHandle : FileSystemHandle {
+  Promise<File> getFile();
+  Promise<FileSystemWritableFileStream> createWritable(optional FileSystemCreateWritableOptions options = {});
+
+  Promise<FileSystemAccessHandle> createAccessHandle();
+  [Exposed=DedicatedWorker]
+  Promise<FileSystemSyncAccessHandle> createSyncAccessHandle();
+};
+
+interface FileSystemAccessHandle {
+  // Assumes seekable streams are available. The
+  // Seekable extended attribute is ad-hoc notation for this proposal.
+  [Seekable] readonly attribute WritableStream writable;
+  [Seekable] readonly attribute ReadableStream readable;
+
+  // Resizes the file to be size bytes long. If size is larger than the current
+  // size the file is padded with null bytes, otherwise it is truncated.
+  Promise<undefined> truncate([EnforceRange] unsigned long long size);
+  // Returns the current size of the file.
+  Promise<unsigned long long> getSize();
+  // Persists the changes that have been written to disk
+  Promise<undefined> flush();
+  // Flushes and closes the streams, then releases the lock on the file
+  Promise<undefined> close();
+};
+
+[Exposed=DedicatedWorker]
+interface FileSystemSyncAccessHandle {
+  unsigned long long read([AllowShared] BufferSource buffer,
+                             FilesystemReadWriteOptions options);
+  unsigned long long write([AllowShared] BufferSource buffer,
+                              FilesystemReadWriteOptions options);
+
+  Promise<undefined> truncate([EnforceRange] unsigned long long size);
+  Promise<unsigned long long> getSize();
+  Promise<undefined> flush(); 
```

Comment:
iirc, in the NativeIO/Storage Foundation API these were all synchronous as well. Could you elaborate on what you learned that made you change the API shape in this newer proposal?